libs/extensions/include/NanoDimuonVertex.hpp

```diff
@@ -15,31 +15,58 @@ typedef Collection<std::shared_ptr<NanoDimuonVertex>> NanoMuonVertices;
 
 class NanoDimuonVertex {
  public:
-  NanoDimuonVertex(std::shared_ptr<PhysicsObject> physicsObject_);
+  NanoDimuonVertex(std::shared_ptr<PhysicsObject> physicsObject_, const std::shared_ptr<Event> event);
 
   auto Get(std::string branchName) { return physicsObject->Get(branchName); }
   float GetAsFloat(std::string branchName) { return physicsObject->GetAsFloat(branchName); }
   std::string GetOriginalCollection() { return physicsObject->GetOriginalCollection(); }
   void Reset() { physicsObject->Reset(); }
 
+  std::shared_ptr<PhysicsObject> Muon1() {return muon1;}
+  std::shared_ptr<PhysicsObject> Muon2() {return muon2;}
+
   bool isDSAMuon1() { return float(physicsObject->Get("isDSAMuon1")) == 1.; }
   bool isDSAMuon2() { return float(physicsObject->Get("isDSAMuon2")) == 1.; }
-  float muonIndex1() { return float(physicsObject->Get("idx1")); }
-  float muonIndex2() { return float(physicsObject->Get("idx2")); }
+  float muonIndex1() { return float(physicsObject->Get("originalMuonIdx1")); }
+  float muonIndex2() { return float(physicsObject->Get("originalMuonIdx2")); }
+  bool isValid() { return (float)Get("isValid") > 0; }
 
   std::shared_ptr<PhysicsObject> GetPhysicsObject() { return physicsObject; }
 
   std::string GetVertexCategory();
   std::pair<std::shared_ptr<PhysicsObject>,std::shared_ptr<PhysicsObject>> GetMuons(const std::shared_ptr<Event> event);
-  float GetDimuonChargeProduct(const std::shared_ptr<Event> event);
 
+  TLorentzVector GetFourVector();
+  float GetInvariantMass() { return GetFourVector().M(); }
+  float GetPt() { return GetFourVector().Pt(); }
+  float GetEta() { return GetFourVector().Eta(); }
+  float GetPhi() { return GetFourVector().Phi(); }
```

Comment:
Since this class represents a vertex, methods like `GetPt` are not immediately obvious to me. Maybe they should be called more like `GetDimuonPt`, etc.?

Reply:
Maybe also do the same for eta and phi?